version: "3.8"

services:
  postgres:
    image: postgres:latest
    container_name: my_postgres
    environment:
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
      POSTGRES_DB: auth_user_db
    ports:
      - "5435:5432"
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U myuser"]
      interval: 10s
      retries: 5
    volumes:
      - ./database.sql:/docker-entrypoint-initdb.d/database.sql
    networks:
      - app_network

  app:
    build:
      context: .
      dockerfile: Dockerfile
    container_name: my_app
    depends_on:
      - postgres
    environment:
      DATABASE_URL: "postgres://myuser:mypassword@postgres:5435/auth_user_db"
    ports:
      - "8080:8080"
    networks:
      - app_network

networks:
  app_network:
    driver: bridge
